src/testsets.jl

```diff
@@ -111,6 +112,37 @@ Return vector containing `rs` so that when iterated through,
 """
 _flatten_results!(rs::Result) = [rs]
 
+"""
+    update_testset_properties!(childts::AbstractTestSet, ts::AbstractTestSet)
+    update_testset_properties!(childts::ReportingTestSet, ts::ReportingTestSet)
+
+Adds properties of `ts` to `childts`. If any properties being added already exist in
+`childts`, a warning is displayed and the value in `childts` is overwritten.
+
+If `ts` and\\or `childts` is not a `ReportingTestSet`, this is handled in the
+`AbstractTestSet` method.
+"""
+function update_testset_properties!(childts::AbstractTestSet, ts::AbstractTestSet)
+    if !isa(childts, ReportingTestSet) && isa(ts, ReportingTestSet) && !isempty(ts.properties)
+        @warn "Properties of testset $(ts.description) can not be added to child testset $(childts.description) as it is not a ReportingTestSet."
+    end
+    # No need to check if childts is ReportingTestSet and ts isn't, as if this is the case
+    # ts has no properties to apply to childts.
+    return childts
+end
+function update_testset_properties!(childts::ReportingTestSet, ts::ReportingTestSet)
+    parent_keys = keys(ts.properties)
+    child_keys = keys(childts.properties)
+    # Loop through keys so that warnings can be issued for any duplicates
+    for key in parent_keys
+        if key in child_keys
+            @warn "Property $key in testest $(childts.description) overwritten by parent testset $(ts.description)"
```

Comment:
Why don't  child  properties take priority over parents?

Reply:
I wasn't which sure which way to do this, and thought it would be more annoying for a parents to be overwritten. Although because of the flattening it'll only overwrite the parent value for the child test set in question, all others child test sets won't be touched. So makes sense to switch, as the child will then have the property you want, and all others won't be changed.